go-controller/vendor/github.com/j-keck/arping/arping_linux.go

```diff
@@ -22,6 +23,9 @@ func initialize(iface net.Interface) (s *LinuxSocket, err error) {
 }
 
 func (s *LinuxSocket) send(request arpDatagram) (time.Time, error) {
+	socketTimeout := timeout.Nanoseconds() * 2
```

Comment:
Using `timeout * 2` both here and in the receive, can make a successful arping round trip top to a duration close to `timeout * 4` which is too wasteful (keeping the file descriptor reserved) given a timeout error would have been reported back to the user after just `timeout`.

So instead of `timeout * 2` I would just use `timeout` both for the send and the receive. That tops at `timeout * 2` which is at least not worse than what we had before.

Reply:
done. Not sure why either, any timeout is better then none, if there where sending issues initially the sender might have never returned